src/libostree/ostree-repo-pull.c

```diff
@@ -1435,7 +1435,7 @@ enqueue_one_object_request (OtPullData        *pull_data,
   expected_max_size_p = (fetchtype != OSTREE_FETCH_OBJECT_CORE) ? NULL : g_hash_table_lookup (pull_data->expected_commit_sizes, checksum);
   if (expected_max_size_p)
     expected_max_size = *expected_max_size_p;
-  else if (is_meta)
+  else if (is_meta && fetchtype != OSTREE_FETCH_OBJECT_COMPAT_SIZES)
```

Comment:
The PR message says that the format could exceed `OSTREE_MAX_METADATA_SIZE`, but here it looks like we’re assigning `OSTREE_MAX_METADATA_SIZE` to `expected_max_size` for `OSTREE_FETCH_COMPAT_OBJECT_SIZES`? I'm not sure I understand what's going on here. I trust that this works since it was in a prior version, but it might be enlightening for me if I knew why the assignment fixed the issue.